BaikalDataGenerator/Source/config_loader.cpp

```diff
@@ -0,0 +1,228 @@
+    ASSERT_PATH(scene_file, "missed scene file");
+    ASSERT_PATH(output_dir, "missed output_dir");
+
+    // validate extansions
+    ASSERT_EXT(camera_file, "camera config should has extenshion '.xml'")
+    ASSERT_EXT(light_file, "light config should has extenshion '.xml'")
+    ASSERT_EXT(spp_file, "spp config should has extenshion '.xml'")
+
+    // validate that files really exists
+    ASSERT_FILE_EXISTING(camera_file, "camera file doesn't exist")
+    ASSERT_FILE_EXISTING(light_file, "light file doesn't exist")
+    ASSERT_FILE_EXISTING(spp_file, "spp file doesn't exist")
+    ASSERT_FILE_EXISTING(scene_file, "scene file doesn't exist")
+
+    if (!std::filesystem::is_directory(output_dir))
+    {
+        THROW_EX("'output_dir' should be directory")
+    }
+}
+
+ConfigLoader::ConfigLoader(const DGenConfig& config)
+{
+    ValidateConfig(config);
+
+    LoadCameraConfig(config.camera_file);
+    LoadLightConfig(config.light_file);
+    LoadSppConfig(config.spp_file);
+}
+
+void ConfigLoader::LoadCameraConfig(const std::filesystem::path& file_name)
+{
+    tinyxml2::XMLDocument doc;
+    doc.LoadFile(file_name.string().c_str());
+
+    auto root = doc.FirstChildElement("cam_list");
+
+    if (!root)
+    {
+        THROW_EX("Failed to open lights set file.")
+    }
+
+    tinyxml2::XMLElement* elem = root->FirstChildElement("camera");
+
+    m_camera_states.clear();
+
+    while (elem)
+    {
+        CameraInfo cam_info;
+
+        // eye
+        cam_info.pos.x = elem->FloatAttribute("cpx");
+        cam_info.pos.y = elem->FloatAttribute("cpy");
+        cam_info.pos.z = elem->FloatAttribute("cpz");
+
+        // center
+        cam_info.at.x = elem->FloatAttribute("tpx");
+        cam_info.at.y = elem->FloatAttribute("tpy");
+        cam_info.at.z = elem->FloatAttribute("tpz");
+
+        // up
+        cam_info.up.x = elem->FloatAttribute("upx");
+        cam_info.up.y = elem->FloatAttribute("upy");
+        cam_info.up.z = elem->FloatAttribute("upz");
+
+        //other values
+        cam_info.focal_length = elem->FloatAttribute("focal_length");
+        cam_info.focus_distance = elem->FloatAttribute("focus_dist");
+        cam_info.aperture = elem->FloatAttribute("aperture");
+
+        m_camera_states.push_back(cam_info);
+        elem = elem->NextSiblingElement("camera");
+    }
+}
+
+void ConfigLoader::LoadLightConfig(const std::filesystem::path& file_name)
+{
+    tinyxml2::XMLDocument doc;
+    doc.LoadFile(file_name.string().c_str());
+    auto root = doc.FirstChildElement("light_list");
+
+    if (!root)
+    {
+        THROW_EX("Failed to open lights set file.")
+    }
+
+    tinyxml2::XMLElement* elem = root->FirstChildElement("light");
+
+    while (elem)
+    {
+        LightInfo light_info;
+        //type
+        light_info.type = elem->Attribute("type");
+
+        // validate light type
+        if (light_info.type == "point") { }
+        else if (light_info.type == "direct") { }
```

Comment:
This empty cases can be removed. And the default case can be replaced by
`else if (light_info.type != "point") && (light_info.type != "point")  THROW...`

Reply:
Done